src/utilities/file.service.ts

```diff
@@ -60,6 +63,50 @@ export class FileService {
         }
     }
 
+    async getNumberOfPdfPages(fileName: string): Promise<number> {
+        const retrievedFile = await this.getFile('resources\\pdfs\\', fileName);
+
+        return getDocument({ data: retrievedFile }).promise.then((document) => document.numPages);
+    }
+
+    async getConvertedPdfFile(fileName: string, pageNumber: number): Promise<StreamableFile> {
+        const outFolderPath = `${process.cwd()}\\out`;
+
+        const options = {
+            format: 'png',
+            out_dir: outFolderPath,
+            out_prefix: 'out',
+            page: pageNumber,
+            scale: 2048,
+        };
+
+        try {
+            return pdf.convert(`${process.cwd()}\\resources\\pdfs\\${fileName}`, options).then(() => {
```

Comment:
We should sanitize the path against directory traversal attacks

Reply:
https://stackoverflow.com/a/38517570 @ErickSharp would be useful to make a method to sanitize everything.